src/android/com/microsoftopentechnologies/intellij/model/storage/TableEntity.java

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright 2014 Microsoft Open Technologies Inc.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.microsoftopentechnologies.intellij.model.storage;
+
+import com.microsoftopentechnologies.intellij.helpers.azure.AzureCmdException;
+import com.microsoftopentechnologies.intellij.model.ServiceTreeItem;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Calendar;
+import java.util.Map;
+import java.util.UUID;
+
+public class TableEntity implements ServiceTreeItem {
+    public enum PropertyType {
+        Boolean,
+        DateTime,
+        Double,
+        Uuid,
+        Integer,
+        Long,
+        String,
+    }
+
+    public static class Property {
```

Comment:
The Property class is actually a TableEntityProperty. We preferred having a nested class with a short name instead of having an independent one with a long name. We believe it's similar to Map and Map.Entry, but we could take it out as an independent class if you prefer it.